ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/AbstractTransactionsLayer.java

```diff
@@ -361,7 +365,12 @@ protected PendingTransaction processRemove(
       final Transaction transaction,
       final RemovalReason removalReason) {
     final PendingTransaction removedTx = pendingTransactions.remove(transaction.getHash());
+
     if (removedTx != null) {
+      if (removedTx.getTransaction().getBlobsWithCommitments().isPresent()
+          && CONFIRMED.equals(removalReason)) {
+        this.blobCache.cacheBlobs(removedTx.getTransaction());
+      }
```

Comment:
Since this only applies when the tx is confirmed, then a best place for it, is in the `confirmed` method, so you do not even need to check the removal reason